flyteplugins/go/tasks/pluginmachinery/tasklog/template.go

```diff
@@ -194,31 +182,50 @@ func (input Input) templateVarsForScheme(scheme TemplateScheme) TemplateVars {
 	return vars
 }
 
+func getDynamicLogLinkTypes(taskTemplate *core.TaskTemplate) []string {
+	if taskTemplate == nil {
+		return nil
+	}
+	config := taskTemplate.GetConfig()
+	if config == nil {
+		return nil
+	}
+	linkType := config["link_type"]
+	if linkType == "" {
+		return nil
+	}
+	return strings.Split(linkType, ",")
+}
+
 func (p TemplateLogPlugin) GetTaskLogs(input Input) (Output, error) {
-	templateVars := input.templateVarsForScheme(p.Scheme)
+	templateVars := input.templateVars()
 	taskLogs := make([]*core.TaskLog, 0, len(p.TemplateURIs))
-
-	// Grab metadata from task template and check if key "link_type" is set to "vscode".
-	// If so, add a vscode link to the task logs.
-	isFlyin := false
-	if input.TaskTemplate != nil && input.TaskTemplate.GetConfig() != nil {
-		config := input.TaskTemplate.GetConfig()
-		if config != nil && config["link_type"] == "vscode" {
-			isFlyin = true
-		}
-	}
 	for _, templateURI := range p.TemplateURIs {
-		// Skip Flyin logs if plugin is enabled but no metadata is defined in input's task template.
-		// This is to prevent Flyin logs from being generated for tasks that don't have a Flyin metadata section.
-		if p.DisplayName == "Flyin Logs" && !isFlyin {
-			continue
-		}
 		taskLogs = append(taskLogs, &core.TaskLog{
 			Uri:           replaceAll(templateURI, templateVars),
 			Name:          p.DisplayName + input.LogName,
 			MessageFormat: p.MessageFormat,
 		})
 	}
 
+	for _, dynamicLogLinkType := range getDynamicLogLinkTypes(input.TaskTemplate) {
+		for _, dynamicTemplateURI := range p.DynamicTemplateURIs {
+			if p.Name == dynamicLogLinkType {
+				for _, match := range dynamicLogRegex.FindAllStringSubmatch(dynamicTemplateURI, -1) {
```

Comment:
nit: Would be nice to support this for the "regular" logs as well...

Reply:
It will be easy to rearrange this logic when we find a good use-case for regular logs.